modules/gateway/peers.go

```diff
@@ -52,8 +53,23 @@ func (g *Gateway) addPeer(p *peer) {
 	go g.listenPeer(p)
 }
 
+// randomPeer returns a random peer from the gateway's peer list.
+func (g *Gateway) randomPeer() (modules.NetAddress, error) {
+	if len(g.peers) > 0 {
+		r := rand.Intn(len(g.peers))
```

Comment:
well, map iteration is pseudo-random as well, so that helps. More importantly, the attacker has no way of knowing the current ordering of your peer list in the first place.